cli/package.json

```diff
@@ -54,7 +54,8 @@
     "eslint-plugin-flowtype": "^2.50.3",
     "eslint-plugin-prettier": "^2.7.0",
     "flow-bin": "^0.81.0",
-    "jest": "^23.6.0"
+    "jest": "^23.6.0",
+    "yarn": "^1.17.3"
```

Comment:
Random passerby comment: Is this intentional? Why would you ever need to install yarn as a dependency (installed via itself), unless your module was using it programmatically (which doesn't appear to be the case)? Looks like an accident?

Reply:
it is intentional and allows you to still use npm scripts (see https://github.com/flow-typed/flow-typed/blob/master/cli/package.json#L16-L25), which for unknown reason depend on yarn, when you don't have global yarn installed.

Reply:
> for unknown reason depend on yarn

This repo uses yarn for development. It has `yarn.lock` files. It [uses `yarn` in the script you're supposed to run to contribute](https://github.com/flow-typed/flow-typed/blob/e6c6a9001fe8627c89a7e363f03f5063b7b153d6/build_and_test_cli.sh#L5-L8) ([reference in CONTRIBUTING.md](https://github.com/flow-typed/flow-typed/blob/e6c6a9001fe8627c89a7e363f03f5063b7b153d6/CONTRIBUTING.md#running-libdef-tests)). By using `npm install` you're bypassing the `yarn.lock`.

Admittedly CONTRIBUTING.md should explicitly mention that this repo uses yarn and you should have it globally installed.

Again I'm just a random passerby who was following an issue, but installing yarn as a local dependency unnecessarily is asking for trouble. Most people will end up with two different versions of yarn. It also fundamentally doesn't make sense, because you need yarn to install yarn this way, if you're doing things correctly, at which point a yarn dev dependency clearly isn't necessary.

Reply:
@serhalp actually, there is no reference of word "yarn" in the whole CONTRIBUTING.md you referenced.

![image](https://user-images.githubusercontent.com/9026581/67310512-8f0b4480-f506-11e9-9799-135ae44cb2eb.png)

the only place in docs about yarn is the README.md, where yarn is used in the example to install flow-typed.

There's no reason why yarn should be used to run npm scripts (see the reference in my previous message). there's no docs stating that you must have `yarn` globally installed to run `npm run test-quick`, so it may confuse new contributors and be somewhat unpleasant.  this package.json change will not install yarn globally, but it ensures that if you don't use yarn, any npm script still should work for you.

Reply:
> actually, there is no reference of word "yarn" in the whole CONTRIBUTING.md you referenced.

That's actually exactly what I said:
> Admittedly CONTRIBUTING.md should explicitly mention that this repo uses yarn and that you should have it globally installed.

It's clearly just missing from there. It's indirectly referenced rather than directly.

I don't know how to explain this any better without repeating myself. There are multiple package managers out there. They are not compatible with each other. All developers on a repo must use the same, blessed package manager. This repo uses yarn, so it has a `yarn.lock`, its `package.json` scripts use `yarn run`, etc. Developers should _not_ used npm. The only issue is this is poorly documented. Specifying `yarn` as a dev dependency which gets installed by `npm` for some `yarn` for others is _not a good practice_. This is hardly subjective.